server/build.gradle.kts

```diff
@@ -76,20 +78,10 @@ tasks.startScripts {
 
 tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().all {
     kotlinOptions {
-        jvmTarget = javaVersion
+        jvmTarget = "11"
```

Comment:
This change reintroduces the issue that was fixed for the Java 17 build in #426 . Quick suggestion to fix it: 
```kotlin
val javaVersion = if (project.hasProperty("javaVersion")) project.property("javaVersion").toString() else "11"
```

Then set `jvmTarget = javaVersion`. This will at least make the Java 17 workflow work without deprecation warnings. 

I see that we specify Java 11 in the shared module as well, but unsure on why that doesn't cause any warnings. Have to have a second look at that.

Reply:
Ok no problem. I've added the ability to override it as you've suggested but with a slight tweak. :)